flloat/base.py

```diff
@@ -29,16 +31,35 @@ def negate(self) -> "Formula":
 
 
 class AtomicFormula(Formula, ABC):
-    """An abstract atomic formula."""
+    """An abstract atomic formula.
 
-    def __init__(self, s: Symbol):
-        """
-        Inintialize the atomic formula.
+    Both formulae and names can be used as atomic symbols.
+    A name must be a string made of letters, numbers, underscores, or it must
+    be a quoted string.
+    """
+
+    name_regex = re.compile(r'(\w+)|(".*")')
+
+    def __init__(self, s: Union[AtomSymbol, Formula]):
+        """Inintializes the atomic formula.
 
-        :param s: the atomic symbol.
+        :param s: the atomic symbol. Formulae are implicitly converted to
+            quoted formulae.
         """
         super().__init__()
-        self.s = s
+        self.s: AtomSymbol
```

Comment:
this won't work in Python 3.6.

Please change it by using `# type: <YourType>`